### PR TITLE
Pass CA cert as secret rather than string in capi-provider-agent

### PR DESCRIPTION
**Pull request by avishayt**

### CODE REVIEW

File: hypershift-operator/controllers/nodepool/nodepool_controller.go [hunk 1]
```diff
@@ -660,9 +660,16 @@ func reconcileUserDataSecret(userDataSecret *corev1.Secret, nodePool *hyperv1.No
 	if err != nil {
 		return fmt.Errorf("failed to marshal ignition config: %w", err)
 	}
-	userDataSecret.Data = map[string][]byte{
-		"disableTemplating": []byte(base64.StdEncoding.EncodeToString([]byte("true"))),
-		"value":             userDataValue,
+	if nodePool.Spec.Platform.Type == hyperv1.AgentPlatform {
```
Comment (enxebre) [inline hypershift-operator/controllers/nodepool/nodepool_controller.go:663]: This is not meant to be used as an API. It's a an impl detail and we won't introduce skew per platform, this is a design invariant.

File: hypershift-operator/controllers/hostedcluster/internal/platform/agent/agent.go [hunk 2]
```diff
@@ -72,13 +60,17 @@ func (Agent) ReconcileSecretEncryption(ctx context.Context, c client.Client, cre
...
 	agentCluster.Spec.ReleaseImage = hcp.Spec.ReleaseImage
 	agentCluster.Spec.ClusterName = hcluster.Name
 	agentCluster.Spec.BaseDomain = hcluster.Spec.DNS.BaseDomain
 	agentCluster.Spec.PullSecretRef = &hcp.Spec.PullSecret
-	if hcluster.Status.IgnitionEndpoint != "" && encodedCACert != "" {
-		agentCluster.Spec.IgnitionEndpoint = &agentv1.IgnitionEndpoint{Url: "https://" + hcluster.Status.IgnitionEndpoint + "/ignition", CaCertificate: encodedCACert}
+
+	caSecret := ignitionserver.IgnitionCACertSecret(hcp.Namespace)
+	if hcluster.Status.IgnitionEndpoint != "" {
+		agentCluster.Spec.IgnitionEndpoint = &agentv1.IgnitionEndpoint{
```
Comment (eranco74) [inline hypershift-operator/controllers/hostedcluster/internal/platform/agent/agent.go:71]: :+1:

**Review by eranco74 — COMMENTED** [review]
/lgtm

**Pull request merged**